flow/project.py

```diff
@@ -4787,7 +4813,7 @@ class MyProject(FlowProject):
 
         # Read the config file and set the internal flag.
         # Do not overwrite with False if not present in config file
-        if flow_config.get_config_value("show_traceback"):
+        if self._flow_config["show_traceback"]:
```

Comment:
Is this being cast correctly to bool? The other config keys were handled with specific casts in `FlowProject.__init__`. We don't want `"False"` (or `"off"` or whatever configobj supports as boolean strings) to be interpreted as truthy.

Reply:
Another good idea. I've largely adapted your snippet as is aside from the fact that we can safely assume that the input is a string since configobj always loads it that way. However, I did uncover a deeper issue with our current config loading: we're not actually doing anything with validation. If validation fails, all that happens is that a [debug-level logger message](https://github.com/glotzerlab/signac/blob/master/signac/common/config.py#L104) is emitted. If a valid value is provided then everything works fine, and configobj will coerce things to the appropriate types as requested by our schema. If an invalid value is provided, though, we're silently passing it through. I've opted not to do anything about it at this time since the immediate next step in the config overhaul is to rewrite this part of signac anyway, but it's something that I wanted documented somewhere.

Reply:
Thanks for the fix, and thanks for looking into this further.